include/BMS/BMS.hpp

```diff
@@ -365,117 +438,677 @@ class BMS {
+            .Data = (uintptr_t) CO_COBID_TPDO_DEFAULT(2) + NODE_ID,
+        },
+        {
+            .Key = CO_KEY(0x1806, 2, CO_UNSIGNED8 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = (uintptr_t) 0xFE,
+        },
+        {
+            .Key = CO_KEY(0x1806, 3, CO_UNSIGNED16 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = (uintptr_t) 0,
+        },
+        {
+            .Key = CO_KEY(0x1806, 5, CO_UNSIGNED16 | CO_OBJ_D__R_),
+            .Type = CO_TEVENT,
+            .Data = (uintptr_t) 2000,
+        },
+        // TPDO7 settings
+        // 0: The TPDO number, default 0
+        // 1: The COB-ID used by TPDO1, provided as a function of the TPDO number
+        // 2: How the TPO is triggered, default to manual triggering
+        // 3: Inhibit time, defaults to 0
+        // 5: Timer trigger time in 1ms units, 0 will disable the timer based triggering
+
+        {
+            .Key = CO_KEY(0x1807, 0, CO_UNSIGNED8 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = (uintptr_t) 7,
+        },
+        {
+            .Key = CO_KEY(0x1807, 1, CO_UNSIGNED32 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = (uintptr_t) CO_COBID_TPDO_DEFAULT(2) + NODE_ID + 1,
+        },
+        {
+            .Key = CO_KEY(0x1807, 2, CO_UNSIGNED8 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = (uintptr_t) 0xFE,
+        },
+        {
+            .Key = CO_KEY(0x1807, 3, CO_UNSIGNED16 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = (uintptr_t) 0,
+        },
+        {
+            .Key = CO_KEY(0x1807, 5, CO_UNSIGNED16 | CO_OBJ_D__R_),
+            .Type = CO_TEVENT,
+            .Data = (uintptr_t) 2000,
+        },
+        // TPDO0 mapping, determines the PDO messages to send when TPDO1 is triggered
+        // 0: The number of PDO messages associated with the TPDO
         // 1: Link to the first PDO message
         // n: Link to the nth PDO message
         {
             .Key = CO_KEY(0x1A00, 0, CO_UNSIGNED8 | CO_OBJ_D__R_),
-            .Type = 0,
-            .Data = (uintptr_t) 1,
+            .Type = nullptr,
+            .Data = (uintptr_t) 5,
         },
         {
             .Key = CO_KEY(0x1A00, 1, CO_UNSIGNED32 | CO_OBJ_D__R_),
-            .Type = 0,
-            .Data = CO_LINK(0x2101, 0, 8),// Link to sample data position in dictionary
+            .Type = nullptr,
+            .Data = CO_LINK(0x2100, 1, 16),
+        },
+        {
+            .Key = CO_KEY(0x1A00, 2, CO_UNSIGNED32 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = CO_LINK(0x2100, 2, 16),
+        },
+        {
+            .Key = CO_KEY(0x1A00, 3, CO_UNSIGNED32 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = CO_LINK(0x2100, 3, 8),
+        },
+        {
+            .Key = CO_KEY(0x1A00, 4, CO_UNSIGNED32 | CO_OBJ_D__R_),
+            .Type = nullptr,
+            .Data = CO_LINK(0x2100, 4, 16),
+        },
```

Comment:
Make sure the link matches the size of the entry it is linked to. These links are defined as 16 bit, but the entry they are linked to are only 8 bit (look at other comment).